processing/src/main/java/io/druid/query/aggregation/last/StringLastFoldingAggregatorFactory.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.last;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import io.druid.java.util.common.StringUtils;
+import io.druid.query.aggregation.Aggregator;
+import io.druid.query.aggregation.BufferAggregator;
+import io.druid.query.aggregation.SerializablePairLongString;
+import io.druid.query.monomorphicprocessing.RuntimeShapeInspector;
+import io.druid.segment.BaseObjectColumnValueSelector;
+import io.druid.segment.ColumnSelectorFactory;
+
+import java.nio.ByteBuffer;
+
+@JsonTypeName("stringLastFold")
+public class StringLastFoldingAggregatorFactory extends StringLastAggregatorFactory
+{
+  public StringLastFoldingAggregatorFactory(
+      @JsonProperty("name") String name,
+      @JsonProperty("fieldName") final String fieldName,
+      @JsonProperty("maxStringBytes") Integer maxStringBytes
+  )
+  {
+    super(name, fieldName, maxStringBytes);
+  }
+
+  @Override
+  public Aggregator factorize(ColumnSelectorFactory metricFactory)
+  {
+    final BaseObjectColumnValueSelector selector = metricFactory.makeColumnValueSelector(getName());
+    return new StringLastAggregator(null, null, maxStringBytes)
+    {
+      @Override
+      public void aggregate()
+      {
+        SerializablePairLongString pair = (SerializablePairLongString) selector.getObject();
+        if (pair != null && pair.lhs >= lastTime) {
+          lastTime = pair.lhs;
+          lastValue = pair.rhs;
+        }
+      }
+    };
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
+  {
+    final BaseObjectColumnValueSelector selector = metricFactory.makeColumnValueSelector(getName());
+    return new StringLastBufferAggregator(null, null, maxStringBytes)
+    {
+      @Override
+      public void aggregate(ByteBuffer buf, int position)
+      {
+        SerializablePairLongString pair = (SerializablePairLongString) selector.getObject();
+        if (pair != null && pair.rhs != null) {
```

Comment:
Same here. `lastTime` should be check even if `pair.rhs` is null.

Reply:
oks!